test/cpp/qps/driver.cc

```diff
@@ -341,6 +342,16 @@ std::unique_ptr<ScenarioResult> RunScenario(
       }
     }
 
+    // Reduce channel count so that total channels specified is held regardless
+    // of the number of clients available
+    size_t num_channels =
+        (i == num_clients - 1)
+            ? client_config.client_channels() - channels_allocated
+            : client_config.client_channels() / num_clients;
```

Comment:
But if this is important, let it be client_channels/(num_clients - i) to avoid bad imbalance.